### PR TITLE
python310Packages.numpyro: 0.12.1 -> 0.13.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Currently the package doesn't build because one of its dependencies (`tensorflow`) is currently broken.
Changelog: https://github.com/pyro-ppl/numpyro/releases/tag/0.13.0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/numpyro/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/numpyro/default.nix:20]:
```suggestion
  disabled = pythonOlder "3.9";
```

> Drops support for python 3.8 and requires jax version >= 0.4.14

**Review by samuela — APPROVED** [review]
diff LGTM happy to merge once `disabled` is updated and nixpkgs-review comes back clean

File: pkgs/development/python-modules/numpyro/default.nix
Comment (GaetanLepage) [reply pkgs/development/python-modules/numpyro/default.nix]: FIxed :)

**Comment by GaetanLepage** [comment]
> diff LGTM happy to merge once `disabled` is updated and nixpkgs-review comes back clean

`nixpkgs-review` is indeed "clean" as `tensorflow` is marked as broken, nothing is to be built.
If you want, we can wait for it to get fixed before merging this.

**Comment by samuela** [comment]
> nixpkgs-review is indeed "clean" as tensorflow is marked as broken, nothing is to be built. If you want, we can wait for it to get fixed before merging this.

nah TF could take a while. as long as numpyro builds I think we should be fine

**Comment by GaetanLepage** [comment]
> diff LGTM happy to merge once `disabled` is updated and nixpkgs-review comes back clean

I checked by manually commenting `tensorflow-probabilities` from the `nativeCheckInputs` and, apart from the failed tests due to the latter missing, `numpyo` builds fine.

So we can go on and merge this PR I guess.

**Pull request merged**
